File: demos/gpio/blink.c
#include <stdio.h>
#include <wiringPi.h>
 
int led_pin = 1;

int main(void)
{
  if (wiringPiSetup() == -1)
    return 1;

  pinMode(led_pin, OUTPUT);
  
  int i;
  for (i = 0; i < 30; ++i)
  {
    digitalWrite(led_pin, 1);
    delay(1000);
    digitalWrite(led_pin, 0);
    delay(1000);
  }

  return 0;
}

File: demos/gpio/blink.js
var wpi = require('wiring-pi');
wpi.setup('wpi');

var led_pin = 1;

wpi.pinMode(led_pin, wpi.OUTPUT);

var value = 1;

var iv = setInterval(function() {
  wpi.digitalWrite(led_pin, value);
  value = +!value;
}, 1*1000);

setTimeout(function () {
  clearInterval(iv);
  wpi.pinMode(led_pin, wpi.INPUT);
}, 60*1000);

